src/core/plugin.go

```diff
@@ -14,3 +14,11 @@ type Plugin interface {
 	Info() *Info
 	Subscriptions() []string
 }
+
+type ExtensionPlugin interface {
+	Init(MessagePipeInterface)
+	Close()
+	Process(*Message)
+	Info() *Info
+	Subscriptions() []string
+}
```

Comment:
There are shorthands for type aliasing if you want:

```suggestion
type ExtensionPlugin interface {
	Plugin
}
```

```suggestion
type ExtensionPlugin = Plugin
```